### DISCUSSION TITLE
Convert pdf file to monochrome

### DISCUSSION THREAD

**Discussion by zain910128** [py-pdf/pypdf#2698] [Q&A] [answered]
How can i convert a file to black and white / monochrome ?

**Comment by pubpub-zz** [comment, marked as answer]
There is no dedicated functions for that. for text and drawings, you should be able to look through contents looking for color operators (CS, cs, SC, SCN, sc, scn, G, g, RG, rg, K, k - see §4.5.7 in pdf 1.7 specification) and replace these lines with G or g operators with the adequate gray definition) but you will have to define your own translation rule.
for images, you can use the `images[ ].replace()` function to replace the image once converted to gray or black

**Comment by zain910128** [comment]
Thanks for your reply.

Is it intended to be added to pypdf as a feature at some point or is it out of scope for this library ?
Can you also point me to any other libraries that you recommend for such a transformation ?

Thanks

**Reply by pubpub-zz** [reply]
pypdf allows to have capabilities to access/modify data. It is not integrate such specific function.

**Reply by zain910128** [reply]
I see. Sometimes we are looking to make changes at page level or file level without going to each part of the page individually. 
Based on what you're saying, it sounds like pypdf is more focussed towards modifying individual elements and parts of a page. Is that right ?
Do you have any recommendation for a library that operates on page level and file level transformations like scale and page size, greyscale colours, rotation, etc ?

**Reply by pubpub-zz** [reply]
pypdf can do page scaling, merging, rotation. for color changing, pypddf can helps you but you need to build your specific code. I know no other library

**Reply by stefan6419846** [reply]
> Is it intended to be added to pypdf as a feature at some point or is it out of scope for this library ?

I would consider this out of scope for *pypdf* as it is too specific, although you are of course invited to share your own implementation under a free license or (depending on the implementation size) provide this as an example for our docs.
